http/src/main/java/com/hazelcast/jet/contrib/http/HttpListenerBuilder.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.contrib.http;
+
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.contrib.http.impl.HttpListenerSourceContext;
+import com.hazelcast.jet.json.JsonUtil;
+import com.hazelcast.jet.pipeline.SourceBuilder;
+import com.hazelcast.jet.pipeline.StreamSource;
+
+import javax.annotation.Nonnull;
+import javax.net.ssl.SSLContext;
+import java.util.Objects;
+
+/**
+ * See {@link HttpListenerSources#builder()}.
+ *
+ * @param <T> the type of the pipeline item.
+ */
+public class HttpListenerBuilder<T> {
+
+    /**
+     * Default port for HTTP(s) listener
+     */
+    public static final int DEFAULT_PORT = 5801;
+
+    private static final int PORT_MAX = 0xFFFF;
+
+    private int port = DEFAULT_PORT;
+    private SupplierEx<SSLContext> sslContextFn;
+    private FunctionEx<byte[], T> mapToItemFn;
+    private Class<T> type;
+
+    HttpListenerBuilder() {
+    }
+
+    /**
+     * Set the port which the source binds and listens.
+     * <p>
+     * For example to bind to port `5802`:
+     * <pre>{@code
+     * builder.port(5802)
+     * }</pre>
+     * <p>
+     * Default value is {@link #DEFAULT_PORT} {@code 5801}.
+     *
+     * @param port the port which the source binds and listens.
+     */
+    @Nonnull
+    public HttpListenerBuilder<T> port(int port) {
```

Comment:
I know it expands the scope, but we should consider the websocket source as well (we already have websocket sink).

Reply:
let's keep the scope of this PR limited for these changes, we can work in other types of sources/sinks in a separate PR